### PR TITLE
indicate to users when an error happens + automatically report

### PR DESCRIPTION
**Pull request by toddmedema**
Depends on https://github.com/ExpeditionRPG/expedition-qdl/pull/19

(so once that gets to deployed to master, we'll need to update package.json here)

Closes https://github.com/ExpeditionRPG/expedition-app/issues/328

### CODE REVIEW

File: app/Main.tsx [hunk 1]
```diff
@@ -178,12 +179,27 @@ function setupGoogleAnalytics() {
...
-  console.log('google analytics set up');
 }
 
 export function init() {
-  // Setup as web platform as default; we might find out later we're an app
   const window = getWindow();
+
+  // Catch and display / log all errors
+  // TODO PROBLEM
+  // Once the parsing error is encountered, it breaks all future attempts to deal with store
```
Comment (smartin015) [inline app/Main.tsx]: Add a reference to #491

File: app/Main.tsx [hunk 2]
```diff
@@ -178,12 +179,27 @@ function setupGoogleAnalytics() {
...
+  // Render, etc
+  window.onerror = function(message: string, source: string, line: number) {
+    const quest = getStore().getState().quest;
+    if (quest && quest.details && quest.details.id) {
+      message = `Quest: ${quest.details.id} - ${quest.details.title}. Error: ${message}`;
+    }
+    const label = (source) ? `${source} line ${line}` : null;
+    logEvent('APP_ERROR', {action: message, label});
+    getStore().dispatch(openSnackbar('Error encounter! Reported automatically.'));
```
Comment (smartin015) [inline app/Main.tsx]: "Error encountered"? and "Reported" might make people think we're reporting them.

Comment (toddmedema) [reply app/Main.tsx]: We are - it's getting automatically logged to Google Analytics, including the quest name + id. That being said, would still be great to engage with the person, at the very least to let them know when it's fixed - updated to `Error! Please report`

File: app/Main.tsx (hunk 1 above)
Comment (toddmedema) [reply app/Main.tsx]: Woops! That's totally not a bug any more. Rm'd comment and closed issue, sorry for the confusion

File: app/Main.tsx [hunk 3]
```diff
@@ -178,12 +179,24 @@ function setupGoogleAnalytics() {
...
+
+  // Catch and display + log all errors
+  window.onerror = function(message: string, source: string, line: number) {
+    const quest = getStore().getState().quest;
+    if (quest && quest.details && quest.details.id) {
+      message = `Quest: ${quest.details.id} - ${quest.details.title}. Error: ${message}`;
+    }
+    const label = (source) ? `${source} line ${line}` : null;
+    logEvent('APP_ERROR', {action: message, label});
+    getStore().dispatch(openSnackbar('Error! Please report.'));
```
Comment (smartin015) [inline app/Main.tsx]: Maybe we can direct them to the "send feedback" button in the app? Otherwise we might just confuse people.

Comment (toddmedema) [reply app/Main.tsx]: Changed to "Please send feedback." and added console.error

**Pull request merged**